src/vim9class.c

```diff
@@ -237,10 +237,14 @@ object_index_from_itf_index(class_T *itf, int is_method, int idx, class_T *cl,
     if (cl == itf)
 	return idx;
 
-    itf2class_T *i2c;
-    for (i2c = itf->class_itf2class; i2c != NULL; i2c = i2c->i2c_next)
-	if (i2c->i2c_class == cl && i2c->i2c_is_method == is_method)
-	    break;
+    itf2class_T *i2c = NULL;
+    for (class_T *super = cl; super != NULL; super = super->class_extends)
+	for (i2c = itf->class_itf2class; i2c != NULL; i2c = i2c->i2c_next)
+	    if (i2c->i2c_class == super && i2c->i2c_is_method == is_method)
+	    {
+		goto found_class_with_interface;
```

Comment:
Instead of a "goto" here, can you use the "break" statement and remove the curly braces?

Reply:
It's a nested for loop. Has "C" learned how to do a break to label?

The reason for the curly braces is that there's a warning without them

Reply:
I overlooked the nested loop.  The other approach is to use a "done" or "success" flag and break out of the nested loops.
This is used in a few places.  The "goto" is used mostly for doing cleanups on error.

Reply:
OK, I guess the complexity and extra instructions to exit the loop wont hurt much.